app/lib/get-workflows-in-order.js

```diff
@@ -0,0 +1,52 @@
+function getWorkflowsInOrder(project, query) {
+  query = query || {};
+  const order = (project.configuration && project.configuration.workflow_order) ?
+    project.configuration.workflow_order :
+    [];
+
+  // TODO remove default page_size once pagination solution implemented
+  query.page_size = query.page_size || 20;
+
+  function getAllWorkflows(query) {
+    let allWorkflows = [];
+    return getWorkflows(query, 1);
+
+    function getWorkflows(query, page) {
+      return project.get('workflows', Object.assign({}, query, { page }))
+        .then((workflows) => {
+          allWorkflows = allWorkflows.concat(workflows);
+          const meta = workflows[0] ? workflows[0].getMeta() : null;
+          if (meta && meta.page !== meta.page_count) {
+            return getWorkflows(query, page + 1);
+          }
+          return Promise.resolve(allWorkflows);
+        })
+        .catch((error) => {
+          console.info(error);
+          return Promise.resolve(allWorkflows);
+        });
+    }
+  }
+
+  return getAllWorkflows(query).then((workflows) => {
+    const workflowsByID = {};
+    workflows.forEach((workflow) => { workflowsByID[workflow.id] = workflow; });
+
+    const workflowsInOrder = order.map(workflowID => workflowsByID[workflowID]).filter(Boolean);
```

Comment:
Should we add sorting by id to the API? It should be pretty cheap as it's the primary key index. The sort order will be maintained throughout the pages.

Reply:
This doesn't sort by ID. It sorts the workflows in the order that's set in the `project.order` array.

Reply:
Gotcha, i see what it's doing now. I miss read that.